near-sdk-core/src/info_extractor/attr_sig_info.rs

```diff
@@ -97,13 +99,27 @@ impl AttrSigInfo {
             }
         }
 
+        let is_view = if let Some(ref receiver) = receiver {
+            let is_view = receiver.mutability.is_none();
+            if is_view && is_payable {
+                return Err(Error::new(
+                    receiver.span(),
+                    "Payable method must be mutable (not view)",
+                ));
+            }
+            is_view
+        } else {
+            false
```

Comment:
Should this be a `view` as well. @nearmax ?

Reply:
I think you're right

Reply:
oh, actually, #init methods is also an exception